### PR TITLE
Fixed test cases with mocks

### PR DESCRIPTION
**Pull request by Euan-lc**
used unittest mocks to fix failing testcases

### CODE REVIEW

**Comment by Amauryeen** [comment]
I am reviewing this PR.

**Comment by Amauryeen** [comment]
✅ The proposed solution holds in one unique commit.
✅ The branch is based on the latest master revision.

✅ The title and the commit description are both written in English.
❎ The title doesn't complete "thanks to this commit we can".
✅ The description describes what's fixed.
❎ The description doesn't reference the issue.

✅ The tests work autonomously and don't depend on `storage`.
✅ The implementation is coherent.

File: tests.py [hunk 1]
```diff
@@ -27,19 +28,25 @@ def setUpClass(cls):
...
-            self.assertEqual(res.status, 404)
+        with patch('simplebin.Snippet') as mock_snippet:
+            mock_snippet.get_by_name.side_effect = ValueError(404)
+            with urlopen('/idontexist') as res:
+                self.assertEqual(res.status, 404)
 
 
 if __name__ == '__main__':
     unittest.main()
+
```
Comment (Julien00859) [inline tests.py:52]: unnecessary diff

File: tests.py [hunk 2]
```diff
@@ -16,6 +16,7 @@ def urlopen(path):
 
 
 class TestSimpleBin(unittest.TestCase):
+
```
Comment (Julien00859) [inline tests.py:19]: unnecessary diff

File: tests.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import urllib.request
 import unittest
 from threading import Thread
-
+from unittest.mock import patch
```
Comment (Julien00859) [inline tests.py:4]: please keep the extra line in between the python standard library imports and the project imports

**Review by Julien00859 — CHANGES_REQUESTED** [review]
Can you squash your three commits into a single one? Keeping the commit message of "Fix the unit test"? Can you also rebase your branch on the latest master?
